Integration of sexaquark simulation code (for EXO-24-003)

lowette opened the pull request:
CMSSW integration of sexaquark simulation code.

This PR adds code for sexaquark simulation starting from HepMC input files. It adds the new sexaquark particles, the inelastic interaction, the neutron annihilation, and a looping mechanism to increase the chance of an interaction, plus other necessary classes and files. CustomPhysicsList is extended to this new interaction for the anti-sexaquark.

Nothing in ordinary CMS running should be affected, GEANT will only use this code when a particle with pdgId -1020000020 is found (which is the proper code for a sexaquark of content uuddss).

This code integration happens following the presentation in the EXO MC&I meeting here:
https://indico.cern.ch/event/1379001/#3-sexaquark-simulation-in-cmss
We were requested to integrate the sexaquark simulation code, and to validate our private productions with a central simulation. For the latter, we will want to do a backport to CMSSW_10_2_X once merged in the master.

The code was tested in the latest 14_1_X integration build using the new python config added in the CustomPhysics/test directory.
This was not expected to affect any runTheMatrix tests. I ran the tests for completeness (runTheMatrix.py -l limited -i all --ibeos) which flagged no problems.

+CC @vetens

lowette committed:
First working version for CMSSW integration of sexaquark simulation code.

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"02825cb8b79afcca0737c758b5cd9188ea10859a":{"files":["SimG4Core/CustomPhysics/data/particles_sexaq_1p5_GeV.txt","SimG4Core/CustomPhysics/data/particles_sexaq_1p7_GeV.txt","SimG4Core/CustomPhysics/data/particles_sexaq_1p85_GeV.txt","SimG4Core/CustomPhysics/data/particles_sexaq_1p8_GeV.txt","SimG4Core/CustomPhysics/data/particles_sexaq_1p9_GeV.txt","SimG4Core/CustomPhysics/data/particles_sexaq_2_GeV.txt","SimG4Core/CustomPhysics/data/particles_sexaq_2p1_GeV.txt","SimG4Core/CustomPhysics/interface/G4AntiSQ.h","SimG4Core/CustomPhysics/interface/G4SQ.h","SimG4Core/CustomPhysics/interface/G4SQInelasticCrossSection.h","SimG4Core/CustomPhysics/interface/G4SQInelasticProcess.h","SimG4Core/CustomPhysics/interface/G4SQLoopProcess.h","SimG4Core/CustomPhysics/interface/G4SQLoopProcessDiscr.h","SimG4Core/CustomPhysics/interface/G4SQNeutronAnnih.h","SimG4Core/CustomPhysics/src/CustomPhysicsList.cc","SimG4Core/CustomPhysics/src/G4AntiSQ.cc","SimG4Core/CustomPhysics/src/G4SQ.cc","SimG4Core/CustomPhysics/src/G4SQInelasticCrossSection.cc","SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc","SimG4Core/CustomPhysics/src/G4SQLoopProcess.cc","SimG4Core/CustomPhysics/src/G4SQLoopProcessDiscr.cc","SimG4Core/CustomPhysics/src/G4SQNeutronAnnih.cc","SimG4Core/CustomPhysics/test/Sexaquark_RunIIFall18GenSim_cfg.py"],"squashed":false,"time":1715896907}},"emoji":{},"last_seen_sha":"02825cb8b79afcca0737c758b5cd9188ea10859a","signatures":{}} -->

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40274

- This PR adds an extra 36KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40274/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40274/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

lowette committed:
Code cleanup after code checks

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40276

- This PR adds an extra 40KB to repository

cmsbuild commented:
A new Pull Request was created by  @lowette for master.

It involves the following packages:

- SimG4Core/CustomPhysics (**simulation**)


@mdhildreth, @cmsbuild, @civanch can you please review it and eventually sign? Thanks.
@rovere, @bsunanda, @slomeo, @makortel, @fabiocos this is something you requested to watch as well.
@rappoccio, @antoniovilela, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

civanch commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39427/summary.html
**COMMIT**: 5804b9c81b64a688e6779374d104f6b55dd1ce84
**CMSSW**: CMSSW_14_1_X_2024-05-17-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44992/39427/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 4 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-17-1100+b1bbb4/62818/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 10 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-17-1100+b1bbb4/62818/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3338976
 - DQMHistoTests: [Total failures: 6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-17-1100+b1bbb4/62818/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3338950
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-17-1100+b1bbb4/62818/triggerResults): no differences found

civanch commented on SimG4Core/CustomPhysics/interface/G4SQInelasticProcess.h:
```diff
@@ -0,0 +1,33 @@
+  G4bool IsApplicable(const G4ParticleDefinition& aParticleType) override;
+
+  // generic PostStepDoIt recommended for all derived classes
+  virtual G4VParticleChange* PostStepDoIt(const G4Track& aTrack, const G4Step& aStep);
```
@lowette , not "virtual" but "override"

lowette replied:
Adapted.

civanch commented on SimG4Core/CustomPhysics/interface/G4SQNeutronAnnih.h:
```diff
@@ -0,0 +1,30 @@
+public:
+  G4SQNeutronAnnih(double mass);
+
+  virtual ~G4SQNeutronAnnih();
```
virtual -> override

lowette replied:
Adapted.

civanch commented on SimG4Core/CustomPhysics/interface/G4SQNeutronAnnih.h:
```diff
@@ -0,0 +1,30 @@
+
+  G4double momDistr(G4double x_in);
+
+  virtual G4HadFinalState* ApplyYourself(const G4HadProjectile& aTrack, G4Nucleus& targetNucleus);
```
virtual -> override

lowette replied:
Adapted.

civanch commented on SimG4Core/CustomPhysics/src/CustomPhysicsList.cc:
```diff
@@ -99,6 +105,22 @@ void CustomPhysicsList::ConstructProcess() {
+          G4SQLoopProcessDiscr* sqLoopPrDiscr = new G4SQLoopProcessDiscr(particle->GetPDGMass() / GeV);
+          pmanager->AddDiscreteProcess(sqLoopPrDiscr);
+        } else if (particle->GetParticleName() == "sexaq") {
+          edm::LogInfo("CustomPhysics") << "   No pmanager implemented for sexaq, only for anti_sexaq";
```
Better to use LogVerbatim instead of LogInfo

lowette replied:
Ok, adapted.

civanch commented on SimG4Core/CustomPhysics/src/G4SQ.cc:
```diff
@@ -0,0 +1,48 @@
+    //             shortlived      subType    anti_encoding
+
+    anInstance = new G4ParticleDefinition(
+        name, mass, 0, 0.0, 0, +1, 0, 0, 0, 0, "baryon", 0, +2, 1020000020, true, -1.0, NULL, false, "sexaq");
```
NULL -> nullptr

lowette replied:
Adapted.

civanch commented on SimG4Core/CustomPhysics/src/G4SQ.cc:
```diff
@@ -0,0 +1,48 @@
+  return theInstance;
+}
+
+//G4SQ* G4SQ::SQDefinition(double mass)
```
If this constructor is not needed these commented lines may be removed.

lowette replied:
Done.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticCrossSection.cc:
```diff
@@ -0,0 +1,45 @@
+G4SQInelasticCrossSection::~G4SQInelasticCrossSection() {}
+
+G4bool G4SQInelasticCrossSection::IsElementApplicable(const G4DynamicParticle* aPart, G4int Z, const G4Material*) {
+  return ((0 < Z) && (aPart->GetDefinition() == theSQ || aPart->GetDefinition() == theAntiSQ));
```
here enough "return true" - this x-section will not be used in any other case.

lowette replied:
Ok, done.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticCrossSection.cc:
```diff
@@ -0,0 +1,45 @@
+  // put the X section low for the antiS to get a flat interaction rate,
+  // but also make it scale with the number of neutrons in the material
+  // because we are going to interact on neutrons, not on protons
+  return (100. * (A - (G4double)Z) / (G4double)Z) * millibarn;
```
@lowette , 1/Z is a really needed factor? What about size of a nucleus?

lowette replied:
I would need to think about this, and didn't change it yet. In a sense, this cross section could be just a constant for now because we only use the beryllium beampipe (for now) and just need the cross section to give a low enough interaction rate so that we can get it more or less flat over the range of the beampipe thickness.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+  theParticle = G4SQ::SQ(mass);
+}
+
+G4SQInelasticProcess::~G4SQInelasticProcess() {}
```
may be "= default"

lowette replied:
Ok changed.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+  theTotalResult->Clear();
+  theTotalResult->Initialize(aTrack);
+  theTotalResult->ProposeWeight(aTrack.GetWeight());
+  if (aTrack.GetTrackStatus() != fAlive) {
```
This cannot happens

lowette replied:
Ok, anyway whole method removed as per your later comment.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+    return theTotalResult;
+  }
+
+  edm::LogInfo("G4SQInelasticProcess::PostStepDoIt") << "Start a possible interaction?" << std::endl;
```
better LogInfo - LogVerbatim, std::endl not needed.

lowette replied:
Ok, anyway whole method removed as per your later comment.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+  edm::LogInfo("G4SQInelasticProcess::PostStepDoIt") << "Start a possible interaction?" << std::endl;
+
+  if (aTrack.GetPosition().rho() / centimeter < 1) {
+    edm::LogInfo("G4SQInelasticProcess::PostStepDoIt")
```
the same

lowette replied:
Ok, anyway whole method removed as per your later comment.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+    G4Exception("G4SQInelasticProcess::PostStepDoIt", "had003", FatalException, ed);
+  }
+
+  // check only for charged particles
```
There is not sense to check after interaction is already sampled.

lowette replied:
Ok, anyway whole method removed as per your later comment.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+
+  // Next check for illegal track status
+  //
+  if (aTrack.GetTrackStatus() != fAlive && aTrack.GetTrackStatus() != fSuspend) {
```
there is no sense in this check also.

lowette replied:
Ok, anyway whole method removed as per your later comment.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+    kineticEnergy /= aParticle->GetParticleDefinition()->GetBaryonNumber();
+
+  try {
+    theInteraction = GetHadronicInteractionList().at(0);
```
there is only one model theInteraction is fixed in this particular case.

civanch commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+        << "In tracker volume? " << (r < (100 * cm) && z < (200 * cm) ? " YES!  " : " NO!  ") << "r=" << r / cm
+        << " z=" << z / cm << std::endl;
+
+    // Check the result for catastrophic energy non-conservation
```
Again: it is enough to make 1 check in your case. LogInfo -> LogVerbatim

lowette replied:
Ok, anyway whole method removed as per your later comment.

civanch commented on SimG4Core/CustomPhysics/src/G4SQNeutronAnnih.cc:
```diff
@@ -0,0 +1,218 @@
+  G4double m_K0S = G4KaonZeroShort::KaonZeroShort()->GetPDGMass();
+  G4double m_L = G4AntiLambda::AntiLambda()->GetPDGMass();
+
+  G4cout << "---->   G4SQNeutronAnnih    <-----" << G4endl;
```
Not protected printout

lowette replied:
Removed.

civanch commented on SimG4Core/CustomPhysics/src/G4SQNeutronAnnih.cc:
```diff
@@ -0,0 +1,218 @@
+  //           << G4endl;
+
+  G4LorentzVector lv1 = aParticle->Get4Momentum();
+  G4cout << "The neutron Fermi momentum (mag, x, y, z) " << targetNucleus.GetFermiMomentum().mag() / MeV << " "
```
Not protected printout

lowette replied:
Changed to LogVerbatim

civanch commented:
@lowette , I think a bit and have a question: why not to use G4HadronicProcess ? process class as a part of this PR practically substitute all everything what G4HadronicProcess is doing in the method PostStepDoIt(..). For debugging it is OK to introduce a new process but now when the PR is nearly ready the process class better to take out.

Another moment, which I do not yet understand: step limitation is applied within beam pipe. Is it because they were looping inside beam pipe and this limitation is needed to make trajectories under control?

civanch commented:
@lowette ,  I see two problems , which is easy to address:

1) class name started for "G4" means that the class is taken from Geant4 distribution. The disclaimer tell us that the Geant4 collaboration design and implement this class. In is not the case, so two actions are needed: all proposed classes "G4XXX" should be renamed to "CMSXXX" or simply into "XXX"; Geant4 disclaimers should be removed. 

2) G4SQInelasticProcess should be renamed to SQInelasticProcess (or CMSSQInelasticProcess), implementation of virtual methods PostStepDoIt() and CheckResults(..) should be removed (base class method will be used), only the constructor and destructor are left in this class. 

1) is needed to avoid a legal conflict: CMSSW should not distribute Geant4 code produced by CMSSW.

2) is more clean and less problems for long-term maintenance. In fact, what is added by original author of this code are debug printouts, which are not needed, because the code is already validated. If needed debugging may be implemented differently without intervention into Geant4 base methods.

lowette committed:
Update for comments from Vladimir Ivantchenko.

lowette commented on SimG4Core/CustomPhysics/src/G4SQInelasticCrossSection.cc:
```diff
@@ -0,0 +1,45 @@
+  if (aPart->GetDefinition() != theAntiSQ)
+    return 0;
+
+  // zero crosssection for particle at rest
```
Ok, removed.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40438

- This PR adds an extra 44KB to repository

- Found files with invalid states:
  - SimG4Core/CustomPhysics/src/G4SQInelasticCrossSection.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQLoopProcessDiscr.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQInelasticCrossSection.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQ.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQLoopProcessDiscr.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQInelasticProcess.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQNeutronAnnih.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQLoopProcess.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQ.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQLoopProcess.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4AntiSQ.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQNeutronAnnih.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4AntiSQ.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40438/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40438/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

lowette commented on SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
```diff
@@ -0,0 +1,241 @@
+  do {
+    try {
+      // Call the interaction
+      result = theInteraction->ApplyYourself(thePro, targetNucleus);
```
Ok, anyway whole method removed as per your later comment.

lowette commented:
Hi @civanch , thanks for the review! I was writing grant proposals so took me time to get back.
I updated the naming of the classes and removed all stray G4 stuff. I also removed the methods in the SQ inelastic process.
Now that I'm writing this I realize you suggest to just go with a default hadronic inelastic process and remove the class CMSSQInelasticProcess altogether? Yes, I think that can work, I'll make the change (maybe tonight) and push another commit for it.

lowette committed:
Removal of the unnecessary intermediate CMSSQInelasticProcess; use directly a G4HadronicProcess instead.

lowette committed:
Code style changes, to please the cmsbuild bot.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44992/40440

- This PR adds an extra 48KB to repository

- Found files with invalid states:
  - SimG4Core/CustomPhysics/src/G4SQInelasticCrossSection.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQLoopProcessDiscr.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQInelasticCrossSection.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQ.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/CMSSQInelasticProcess.h:
    - Added: 2023651129b3b85a5711ec743abeadb3b4ce1cef
    - Deleted: 20cf0587c802007767ca526f1338e54ba8f7dbf5
  - SimG4Core/CustomPhysics/interface/G4SQLoopProcessDiscr.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/CMSSQInelasticProcess.cc:
    - Added: 2023651129b3b85a5711ec743abeadb3b4ce1cef
    - Deleted: 20cf0587c802007767ca526f1338e54ba8f7dbf5
  - SimG4Core/CustomPhysics/interface/G4SQInelasticProcess.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQNeutronAnnih.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQLoopProcess.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQ.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4SQLoopProcess.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/interface/G4AntiSQ.h:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4SQNeutronAnnih.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef
  - SimG4Core/CustomPhysics/src/G4AntiSQ.cc:
    - Added: 02825cb8b79afcca0737c758b5cd9188ea10859a
    - Modified: 5804b9c81b64a688e6779374d104f6b55dd1ce84
    - Deleted: 2023651129b3b85a5711ec743abeadb3b4ce1cef

cmsbuild commented:
Pull request #44992 was updated. @mdhildreth, @civanch, @cmsbuild can you please check and sign again.

civanch commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/summary.html
**COMMIT**: 8e4943c6d73a06e5424ba3d8e2457b989757b3f3
**CMSSW**: CMSSW_14_1_X_2024-05-30-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44992/39637/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/4.53_RunPhoton2012B)```4.53_RunPhoton2012B/step2_RunPhoton2012B.log```
- [7.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/7.3_CosmicsSPLoose2018)```7.3_CosmicsSPLoose2018/step1_CosmicsSPLoose2018.log```
- [8.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/8.0_BeamHalo)```8.0_BeamHalo/step1_BeamHalo.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/9.0_Higgs200ChargedTaus)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/25.0_TTbar)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/136.731_RunSinglePh2016B)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/139.001_RunMinimumBias2021)
- [4.22](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/4.22_RunCosmics2011A)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/136.874_RunEGamma2018C)
- [140.023](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/140.023_RunZeroBias2022B)
- [140.043](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/140.043_RunZeroBias2022C)
- [140.063](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/140.063_RunZeroBias2022D)
- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/135.4_ZEEFS_13)
- [5.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/5.1_TTbarFS)
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/136.88811_RunJetHT2018DreMINIAODUL)
- [141.042](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/141.042_RunZeroBias2023D)
- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/136.7611_RunJetHT2016EreMINIAOD)
- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/136.8311_RunJetHT2017FreMINIAOD)
- [141.044](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/141.044_RunJetMET2023D)
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/1306.0_SingleMuPt1_UP15)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/1330.0_ZMM_13)
- [141.046](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/141.046_RunEGamma2023D)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/25202.0_TTbar_13)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2022)
- [101.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/101.0_SingleElectronE120EHCAL)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/140.56_RunHI2018)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [12834.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/12834.7_TTbar_14TeV+2024_trackingMkFit)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/12834.0_TTbar_14TeV+2024)
- [12846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/12846.0_ZEE_14+2024)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/13034.0_TTbar_14TeV+2024PU)
- [158.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/158.01_HydjetQB12ppRECOin2018reMINIAOD)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/13234.0_TTbar_14TeV+2021FS)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/14034.0_TTbar_14TeV+2023FS)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/14234.0_TTbar_14TeV+2023FSPU)
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/24834.0_TTbar_14TeV+2026D98)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/23234.0_TTbar_14TeV+2026D94)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
- [25034.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/1001.0_RunMinBias2011A)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/1000.0_RunMinBias2011A)
- [2500.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/runTheMatrix-results/2500.4_NANOmc132X)
</details>

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b1bbb4/39637/summary.html
**COMMIT**: 8e4943c6d73a06e5424ba3d8e2457b989757b3f3
**CMSSW**: CMSSW_14_1_X_2024-05-30-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44992/39637/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



There are some workflows for which there are errors in the baseline:
 1000.0 step 2
1001.0 step 2
101.0 step 1
10224.0 step 1
11634.0 step 1
12434.0 step 1
12834.0 step 1
12834.7 step 1
12846.0 step 1
13034.0 step 1
1306.0 step 1
13234.0 step 1
1330.0 step 1
135.4 step 1
136.731 step 2
136.7611 step 2
136.793 step 2
136.8311 step 2
136.874 step 2
136.88811 step 2
139.001 step 2
140.023 step 2
140.043 step 2
140.063 step 2
140.56 step 2
14034.0 step 1
141.042 step 2
141.044 step 2
141.046 step 2
14234.0 step 1
158.01 step 2
23234.0 step 1
24834.0 step 1
24834.911 step 1
24896.0 step 1
24900.0 step 1
25.0 step 1
2500.4 step 2
250202.181 step 1
25034.999 step 1
25202.0 step 1
312.0 step 1
4.22 step 2
4.53 step 2
5.1 step 1
7.3 step 1
8.0 step 1
9.0 step 1
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

**Summary**:
 - You potentially [added 19181 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-30-2300+b1bbb4/62999/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-30-2300+b1bbb4/62999/validateJR.html)
 - DQMHistoTests: Total files compared: 1
 - DQMHistoTests: Total histograms compared: 0
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-30-2300+b1bbb4/62999/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 0
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 66 log files, 0 edm output root files, 1 DQM output files

civanch commented:
@smuzaffar , sorry, I cannot interpret tests as "approved". This does not related to this PR but bot report is not clear to me.

smuzaffar commented:
please test

lets re-run to get clean tests: see https://github.com/cms-sw/cmssw/issues/45116

civanch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @rappoccio, @sextonkennedy, @antoniovilela (and backports should be raised in the release meeting by the corresponding L2)

rappoccio commented:
+1
